src/platform/ESP32/OTAImageProcessorImpl.cpp

```diff
@@ -142,13 +303,31 @@ void OTAImageProcessorImpl::HandlePrepareDownload(intptr_t context)
         ChipLogError(SoftwareUpdate, "OTA partition not found");
         return;
     }
+#ifdef CONFIG_ENABLE_DELTA_OTA
+    // New image size is unknown for delta OTA, so we use OTA_SIZE_UNKNOWN flag.
+    esp_err_t err = esp_ota_begin(imageProcessor->mOTAUpdatePartition, OTA_SIZE_UNKNOWN, &(imageProcessor->mOTAUpdateHandle));
+#else
     esp_err_t err =
         esp_ota_begin(imageProcessor->mOTAUpdatePartition, OTA_WITH_SEQUENTIAL_WRITES, &(imageProcessor->mOTAUpdateHandle));
+#endif // CONFIG_ENABLE_DELTA_OTA
+
     if (err != ESP_OK)
     {
         imageProcessor->mDownloader->OnPreparedForDownload(ESP32Utils::MapError(err));
         return;
     }
+#ifdef CONFIG_ENABLE_DELTA_OTA
+    imageProcessor->deltaOtaCfg.user_data               = imageProcessor,
+    imageProcessor->deltaOtaCfg.read_cb                 = &(imageProcessor->DeltaOTAReadCallback),
+    imageProcessor->deltaOtaCfg.write_cb_with_user_data = &(imageProcessor->DeltaOTAWriteCallback),
+
+    imageProcessor->mDeltaOTAUpdateHandle = esp_delta_ota_init(&imageProcessor->deltaOtaCfg);
+    if (imageProcessor->mDeltaOTAUpdateHandle == NULL)
+    {
+        ChipLogError(SoftwareUpdate, "esp_delta_ota_init failed");
+        return;
```

Comment:
We should notify the waiting API that we failed...

```suggestion
        ChipLogError(SoftwareUpdate, "esp_delta_ota_init failed");
        imageProcessor->mDownloader->OnPreparedForDownload(CHIP_ERROR_INTERNAL);
        return;
```